main.cpp

```diff
@@ -360,12 +362,111 @@ difficulty_type next_difficulty_v3_old(std::vector<uint64_t> timestamps, std::ve
 	return next_D;
 }
 
-int main(int argc, char **argv) 
+difficulty_type next_difficulty_v4_interp(std::vector<uint64_t> timestamps, std::vector<difficulty_type> cumulative_difficulties)
 {
-	uint64_t timestamp = 1500000000;
+        if(timestamps.size() != N + 1 || cumulative_difficulties.size() != N + 1)
+                abort();
+
+		// the newest timestamp is not allwed to be older than the previous
+		uint64_t t_last = std::max(timestamps[N], timestamps[N - 1] - 0 * T);
+		// the newest timestamp can only be 3 target times in the future
+        timestamps[N] = std::min(t_last, timestamps[N - 1] + 3 * T);
```

Comment:
I suggest to set `3 * T` to `2 * T` which is similar to ignore the newest block time. This will reduce to strong reactions on faked timestamps.